src/com/lilithsthrone/utils/Util.java

```diff
@@ -524,41 +565,9 @@ public static String addStutter(String sentence, int frequency) {
 	 *            modified sentence
 	 */
 	public static String addBimbo(String sentence, int frequency) {
-		splitSentence = sentence.split(" ");
+		sentence = insertIntoSentence(sentence, frequency, bimboWords);
 		utilitiesStringBuilder.setLength(0);
-
-		// 1 in "frequency" words are bimbo interjections, with a minimum of 1.
-		int wordsToBimbofy = splitSentence.length / frequency + 1;
-
-		int offset = 0;
-		for (int i = 0; i < wordsToBimbofy; i++) {
-			offset = random.nextInt(frequency);
-			offset = ((i * frequency + offset) >= splitSentence.length ? splitSentence.length - 1 : (i * frequency + offset));
-			if (offset != 0) {
-				// If previous word didn't end with punctuation:
-				if (splitSentence[offset - 1].charAt(splitSentence[offset - 1].length() - 1) != '.' && splitSentence[offset - 1].charAt(splitSentence[offset - 1].length() - 1) != '!'
-						&& splitSentence[offset - 1].charAt(splitSentence[offset - 1].length() - 1) != '?') {
-					// Add a comma to the end of the previous word:
```

Comment:
Unless I'm missing something, the previous-word comma added here isn't in the new function.